### DISCUSSION TITLE
Running otvdm on ARM/Parallels…nearly there!

### DISCUSSION THREAD

**Discussion by Ralphster45** [otya128/winevdm#1377] [Ideas]
Hi,

I have read the other posts about running otvdm with Windows 11 on parallels with a MacBook M1 (ARM processor).

Like others, I find that if I try to install otvdm on the Mac, I get the Install error:

"The INF file you selected does not support this method of installation"

So I am unable to do a full install of otvdm.

I am trying to install Visual Basic 6 on the Mac. Unfortunately, the VB6 setup calls a 16 bit executable called ACMBOOT.exe which, of course, fails under W11.

I have been able to launch ACMBOOT.exe by running otvdmw.exe and then selecting ACMBOOT.exe. This is brilliant, and shows I am within sight of the finishing line, however the ACMBOOT runs a bit, and then gets upset, presumably because it is running independently and wasn't properly called by the setup.

It seems I am tantalising close, but I need somehow, to get otvdm properly installed so that ACMBOOT can run as and when it is called by the VB6 setup.

Is there a manual process I can do to install otvdm so that 16 bit programs will run autonomously, just to get me past this snag?

I don't mind if it's long and tedious, I should only need to do it once.

Many thanks for your help with this

Cheers

Ralph

**Comment by emendelson** [comment]
This is a frustrating problem. I tried adding the relevant registry keys to different parts of the registry, including the `WOWAA32Node` branch, but with no effect. If anyone has better ideas, it would be good to know.

**Comment by otya128** [comment]
The ARM64 version of Windows does not contain the necessary DLL, so rewriting the registry will not work.
It can be installed using https://github.com/otya128/ntvdm64.dll (A subset implementation of the necessary DLL)

**Comment by emendelson** [comment]
Thank you! I didn't know about that. I've updated my Inno Setup installer so that it installs the DLL under ARM-based Windows 11 and writes the correct entries to the desktop. Scroll down this page to the paragraph about the Inno Setup installer (and make sure to read the paragraph elsewhere on the page that begins "Virus detection"):

https://mendelson.org/otvdm.html

EDIT: This should now work with Windows 10 on Intel and ARM. The previous version of the installer ran only under Windows 11.

**Comment by Ralphster45** [comment]
Just a quick note from me.

Thanks very much to otya128 for otvdm and also to emendelson for the installer, and extra help via PMs.

This problem is now sorted as per the above posts.

ATB Ralph
